### PR TITLE
Get working on 1.0

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

**Comment by BeastyBlacksmith** [comment]
@simonbyrne can you tell, what is still missing / where I can help?

**Comment by simonbyrne** [comment]
Okay, this is a fairly significant update/rewrite. I've gotten rid of the exported macros, and it no longer operates directly on Julia `Expr`s and `Symbol`s, but instead defines its own `WExpr`/`WSymbol` types. For now the interface is mostly based around `W""` expressions and the `weval` function, but I could see adding more functionality (e.g. a `@weval` function that can operate on Julia expressions).

While there is still more I would like to do, I feel this is working sufficiently well that it could be merged. If you feel this is too much change, I could also move it to a new package.

Any thoughts @BeastyBlacksmith, @MikeInnes?

**Comment by BeastyBlacksmith** [comment]
would be nice to have `convert(Expr, WExpr)`, but that can still be added later or in `Mathematica.jl`/ another package.
Good to see it back alive :+1:

**Comment by BeastyBlacksmith** [comment]
Maybe the README could be clearer on what to do exactly. Here is what I did:
I searched for `MathKernel` and `libWSTP64i4.so` in my Mathematica installation and then assigned them to `ENV["JULIA_MATHKERNEL"]` and `ENV["JULIA_MATHLINK"]`.
Then the package builds, but running `weval` I get 
```
ERROR: InitError: ccall: could not find function MLInitialize in library /usr/local2/app/Wolfram/Mathematica/12.0.0/SystemFiles/Links/WSTP/DeveloperKit/Linux-x86-64/CompilerAdditions/libWSTP64i4.so
```

File: deps/build.jl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        end
+
+    elseif Sys.isunix()
+        archdir = Sys.ARCH == :arm ?    "Linux-ARM" :
+                  Sys.ARCH == :x86_64 ? "Linux-x86-64" :
+                                        "Linux"
+
+        # alternatively, "math" or "wolfram" is often in PATH, so could use
+        # echo \$InstallationDirectory | math | sed -n -e 's/Out\[1\]= //p'
+
```
Comment (BeastyBlacksmith) [inline deps/build.jl]: couldn't we use `mpath =  run(`which mathematica`)` ?

**Comment by BeastyBlacksmith** [comment]
Got it working.

File: deps/build.jl (hunk 1 above)
Comment (simonbyrne) [reply deps/build.jl]: That's a good idea.

**Comment by simonbyrne** [comment]
Okay, I'm pulling the trigger on this. Please feel free to open issues/PRs for the above points.

**Pull request merged**